mlem/contrib/xgboost.py

```diff
@@ -116,11 +102,33 @@ def process(cls, obj: xgboost.DMatrix, **kwargs) -> DataType:
     def get_writer(
         self, project: str = None, filename: str = None, **kwargs
     ) -> DataWriter:
-        raise NotImplementedError()
+        raise NotImplementedError
 
-    def get_model(self, prefix: str = "") -> Type[BaseModel]:
+
+class DMatrixSerializer(DataSerializer[DMatrixDataType]):
+    """Serialzier for XGBoost DMatrix"""
+
+    is_default: ClassVar = True
+    data_class: ClassVar = DMatrixDataType
+
+    def get_model(self, data_type, prefix: str = "") -> Type[BaseModel]:
         raise NotImplementedError
 
+    def serialize(
```

Comment:
Minor peeve: Need to implement (inherited `NotImplementedError`)?
Maybe the `NotImplementedError` can just serialize the class name in the error message and than this wouldn't be needed